### PR TITLE
Update to use latest package:vm_service 2.2.0

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Some classes in the latest vm_service package have new required parameters (debugger / memory related classes). I was unclear on what to add for most of them, so reviewers please take a close look at these and comment if you think the values should be different.

@terrylucas @DaveShuckerow 

@bkonyi the change of the `setFlag` RPC to return `Future<dynamic>` instead of `Future<Success>` seems like it might be a regression. Was this change intentional?

### CODE REVIEW

File: packages/devtools_app/lib/src/debugger/callstack_view.dart [hunk 1]
```diff
@@ -61,9 +61,14 @@ class CallStackView implements CoreElementView {
   void showFrames(List<Frame> frames, {bool selectTop = false}) {
     if (frames.isEmpty) {
       // Create a marker frame for 'no call frames'.
-      final Frame frame = Frame()
-        ..kind = emptyStackMarker
-        ..code = (CodeRef()..name = '<no call frames>');
+      final Frame frame = Frame(
+        index: 0,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/callstack_view.dart]: new required param

File: packages/devtools_app/lib/src/debugger/html_debugger_screen.dart [hunk 2]
```diff
@@ -321,18 +321,22 @@ class HtmlDebuggerScreen extends HtmlScreen {
...
           final Frame frame = frames.first;
 
-          final Frame newFrame = Frame()
-            ..type = frame.type
-            ..index = frame.index
-            ..function = frame.function
-            ..code = frame.code
-            ..location = frame.location
-            ..kind = frame.kind;
+          final Frame newFrame = Frame(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/html_debugger_screen.dart]: type param removed

File: packages/devtools_app/lib/src/debugger/html_debugger_screen.dart [hunk 3]
```diff
@@ -321,18 +321,22 @@ class HtmlDebuggerScreen extends HtmlScreen {
...
           final List<BoundVariable> newVars = <BoundVariable>[];
-          newVars.add(BoundVariable()
-            ..name = '<exception>'
-            ..value = reportedException);
+          newVars.add(BoundVariable(
+            name: '<exception>',
+            value: reportedException,
+            scopeStartTokenPos: null,
+            scopeEndTokenPos: null,
+            declarationTokenPos: null,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/html_debugger_screen.dart]: new required params here and elsewhere

File: packages/devtools_app/lib/src/memory/html_memory_data_view.dart [hunk 4]
```diff
@@ -130,19 +130,35 @@ class MemoryDataChildProvider extends ChildProvider<BoundField> {
...
-                ..decl = (FieldRef()..name = '[$index]')
-                ..value = value);
+              result.add(BoundField(
+                decl: FieldRef(
+                  name: '[$index]',
+                  isStatic: null,
+                  owner: null,
+                  isFinal: null,
+                  declaredType: null,
+                  isConst: null,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/memory/html_memory_data_view.dart]: new required params here and elsewhere

File: packages/devtools_app/lib/src/profiler/cpu_profile_service.dart [hunk 5]
```diff
@@ -27,7 +27,7 @@ class CpuProfilerService {
         .clearCpuSamples(serviceManager.isolateManager.selectedIsolate.id);
   }
 
-  Future<Success> setProfilePeriod(String value) {
+  Future<dynamic> setProfilePeriod(String value) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/profiler/cpu_profile_service.dart]: @bkonyi is this intentional (`Success -> dynamic`)

Comment (bkonyi) [reply packages/devtools_app/lib/src/profiler/cpu_profile_service.dart]: Yes, see https://github.com/dart-lang/sdk/commit/35748514c724a3da9c45d8b45b580f3594eb179e

**Comment by bkonyi** [comment]
FYI, the required parameters match those that are not listed as `optional` in the service specification. You should be safe to provide dummy values when creating most of these, assuming you're not passing them back to the service.

**Pull request merged**